### PR TITLE
Security/129 implement rate limiting for video upload endpoint

### PR DESCRIPTION
**Pull request by alyall17**
Rate limiting added to Flask API upload endpoints to restrict number of uploads allowed per user & per IP address by integrating Flask-Limited into backend, configuring rate limits for video upload endpoint, adding logging for rate limit violations and updating API documentation to reflect upload limits. Unit testing added to ensure current functionality is not broken

### CODE REVIEW

File: app/exercises/routes.py [hunk 1]
```diff
@@ -83,7 +93,19 @@ def parse_user_video(video_file, exercise, user_id):
...
+        return f"user:{user_id}"
+
+    forwarded_for = (request.headers.get('X-Forwarded-For') or '').split(',')[0].strip()
+    remote_ip = forwarded_for or request.remote_addr or 'unknown'
+    return f"anon:{remote_ip}"
+
+
 @exercises_bp.route('/analyze', methods=['POST'])
+@limiter.limit(_get_cv_upload_ip_rate_limit, key_func=get_remote_address)
+@limiter.limit(_get_cv_upload_user_rate_limit, key_func=_upload_user_rate_limit_key)
```
Comment (aryansri0208) [inline app/exercises/routes.py:108]: Nice use of both IP + user limits. Consider adding a short comment explaining why both are applied.

File: app/exercises/routes.py [hunk 2]
```diff
@@ -25,6 +28,13 @@
 EXERCISES_DIR = os.path.dirname(os.path.abspath(__file__))
 VIDEO_IN_DIR = os.path.join(EXERCISES_DIR, 'video_in')
 
+def _get_cv_upload_user_rate_limit():
+    return os.getenv('CV_UPLOAD_USER_RATE_LIMIT', '10 per minute')
```
Comment (aryansri0208) [inline app/exercises/routes.py:32]: Default values are hardcoded — consider moving these to a config/constants file for better visibility and reuse.

File: app/rate_limit.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        request.path,
+        request.method,
+        user_id,
+        ip_address,
+        str(error.description),
+    )
+    return jsonify({
+        "error": "Rate limit exceeded",
+        "detail": str(error.description),
+    }), 429
```
Comment (aryansri0208) [inline app/rate_limit.py:38]: Good structured response — consider also documenting returned rate-limit headers

**Review by aryansri0208 — COMMENTED** [review]
Clean and well-structured implementation with good test coverage and thoughtful use of user + IP rate limiting, just needs minor scalability and edge-case improvements.
